### PR TITLE
Create contributors guide

### PR DESCRIPTION
**Pull request by matthieucan**
While there, fix some inconsistencies in the documentation website, and update the readme.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -81,10 +81,10 @@ Every rule can be configured with the following option:
   of 4.
 
 Some rules have additional configuration options, e.g.
-[sql_has_reasonable_number_of_lines](/rules/generic/#sql_has_reasonable_number_of_lines).
+[sql_has_reasonable_number_of_lines](rules/generic.md#sql_has_reasonable_number_of_lines).
```
Comment (jochemvandooren) [inline docs/configuration.md:84]: Great, I noticed these links as well :ok_hand:

File: README.md [hunk 2]
```diff
@@ -7,92 +7,34 @@
...
+constraints, 3rd party integrations... Not to mention any other metadata that
+organizations need, fully supported through the `meta` parameter.
 
-- Any Python version starting from 3.11
-- [pre-commit](https://pre-commit.com/)
-- [PDM](https://pdm-project.org/2.12/)
+At scale, with hundreds or thousands of data models, all this metadata can
+become confusing, disparate, and inconsistent. It's hard to enforce good
+practices and maintain them in continuous integration systems. This is where
+`dbt-score` chimes: by allowing data teams to programatically define and enforce
```
Comment (jochemvandooren) [inline README.md:28]: `chimes`:

Never heard of this, also unable to find a good definition :thinking:

Comment (jochemvandooren) [reply README.md:28]: shines?

File: docs/contributors_guide.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+### Fixing bugs
+
+We love bug squashing! You can open a pull request to fix any bug you encounter
+in `dbt-score`. If the changes are large enough, refer to the next section
+first - discussing a solution in a Github issue is always a good idea to avoid
+unnecessary work and orchestrate efforts.
+
+### Adding or changing core features
+
+Before implementing or changing a new feature, we kindly ask you to
```
Comment (jochemvandooren) [inline docs/contributors_guide.md:44]: I think we mention a lot of times that they should always open an issue first, becomes a little bit too much IMO

File: docs/create_rules.md [hunk 4]
```diff
@@ -2,8 +2,9 @@
 
 In order to lint and score models, `dbt-score` uses a set of rules that are
 applied to each model. A rule can pass or fail when it is run. Based on the
-severity of the rule models are scored. `dbt-score` has a set of rules enabled
-by default, which can be found [here](reference/rules/generic.md).
+severity of the rule, models are scored with the weighted average of the rules
```
Comment (jochemvandooren) [inline docs/create_rules.md:5]: It's a little bit more complicated than the weighted average, right?

File: docs/index.md [hunk 5]
```diff
@@ -10,8 +10,8 @@ encourage) good practices.
 
 ## Example
 
-```shell
-$ dbt-score lint
+```
+> dbt-score lint
```
[the remark this answers is not in the document]
Comment (jochemvandooren) [reply docs/index.md:14]: It is, good change :ok_hand:

**Review by jochemvandooren — COMMENTED** [review]
All in all looking good! :rocket: Just a couple of comments

**Review by druzhinin-kirill — APPROVED** [review]
IMO looks good to begin with 👍

File: README.md (hunk 2 above)
Comment (matthieucan) [reply README.md:28]: Nice catch, I meant "chimes in" (https://dictionary.cambridge.org/dictionary/english/chime-in) but will replace with "plays its role", it's more clear

File: docs/create_rules.md (hunk 4 above)
Comment (matthieucan) [reply docs/create_rules.md:5]: There's the rule of "critical -> 0", but other than this it's a weighted average, right? Not sure how to express this in a simple manner for the readme

File: README.md (hunk 2 above)
Comment (jochemvandooren) [reply README.md]: :+1:

File: docs/create_rules.md (hunk 4 above)
Comment (jochemvandooren) [reply docs/create_rules.md:5]: Ok let's keep it like this! :+1:

**Review by jochemvandooren — APPROVED** [review]
:rocket:

**Pull request merged**